ruby/ql/lib/codeql/ruby/experimental/Rbi.qll

```diff
@@ -0,0 +1,426 @@
+/**
+ * Provides classes and predicates for working with Ruby Interface (RBI) files
+ * and concepts. RBI files are valid Ruby files that can contain type
+ * information used by Sorbet for typechecking.
+ */
+
+private import codeql.ruby.ApiGraphs
+private import codeql.ruby.AST
+private import codeql.ruby.CFG
+private import codeql.ruby.controlflow.CfgNodes
+
+/**
+ * Provides classes and predicates for working with Ruby Interface (RBI) files
+ * and concepts. RBI files are valid Ruby files that can contain type
+ * information used by Sorbet for typechecking.
+ */
+module Rbi {
+  /**
+   * Contains classes representing RBI types.
+   */
+  private module RbiTypes {
+    /**
+     * A node representing a Ruby Interface (RBI) type.
+     */
+    abstract class RbiType extends Expr { }
+
+    class ConstantReadAccessAsRbiType extends RbiType {
```

Comment:
This class has no qldoc and its name isn't very descriptive. Is it supposed to represent a named class or module type?

Reply:
Forgot to add qldoc for this. It's intended to represent cases like read accesses to the `Integer` and `MyList` classes and the `MyList2` constant in:
```
  MyList2 = T.type_alias(MyList)

  sig { params(l: MyList2).returns(Integer) }
  def len(l); end
```

Generally I think that these types would always represent some class, possibly via some intermediate type aliases, but the class definitions might be not be available in the database (`Integer` and `String` are common examples). I ended up making this class very broad to avoid missing cases like these.

Reply:
Separately, I've noticed that I can simplify this as just `class ConstantReadAccessAsRbiType extends RbiType, ConstantReadAccess { }` after moving this library to the AST layer.